### PR TITLE
fix(flink): rewrite `ops.Date` as a cast operation + fix(flink): fix translation of `ops.TimestampDiff`

### PR DESCRIPTION
**Pull request by deepyaman**

### CODE REVIEW

File: ibis/backends/flink/registry.py [hunk 1]
```diff
@@ -63,18 +72,20 @@ def _filter(translator: ExprTranslator, op: ops.Node) -> str:
...
+def _timestamp_from_unix(translator: ExprTranslator, op: ops.Node) -> str:
+    arg, unit = op.args
 
-def _nullifzero(translator: ExprTranslator, op: ops.Literal) -> str:
-    casted = translate_literal(ops.Literal("0", dtype=op.dtype))
-    return f"NULLIF({translator.translate(op.arg)}, {casted})"
+    if unit == TimestampUnit.MILLISECOND:
+        return f"TO_TIMESTAMP_LTZ({helpers.quote_identifier(arg.name, force=True)}, 3)"
+    elif unit == TimestampUnit.SECOND:
+        return f"TO_TIMESTAMP_LTZ({helpers.quote_identifier(arg.name, force=True)}, 0)"
```
Comment (jcrist) [inline ibis/backends/flink/registry.py:87]: Nit - these lines look identical except for the 3/0 as the 2nd arg. Could you refactor to reduce the code duplication?

**Review by jcrist — COMMENTED** [review]
One tiny nit, otherwise LGTM

File: ibis/backends/flink/registry.py (hunk 1 above)
Comment (deepyaman) [reply ibis/backends/flink/registry.py]: Done! I think it probably copies something similar from, say, `mssql` backend.

Comment (deepyaman) [reply ibis/backends/flink/registry.py]: (But it didn't feel clear enough value to reduce the duplication on that backend, from what I saw.)

Comment (jcrist) [reply ibis/backends/flink/registry.py]: Thanks! And yeah, no need to fixup existing code now; just trying to do small cleanups as new code is added.

**Review by jcrist — APPROVED** [review]

**Pull request merged**